src/DeribitSolution/Program.cs

```diff
@@ -21,19 +20,25 @@
 configuration = new ConfigurationBuilder()
     .SetBasePath(path)
     .AddJsonFile("appsettings.json", optional: false, reloadOnChange: true)
+    .AddUserSecrets("c252577f-cf0f-45f3-99b3-4243ba11d930")
     .Build();
 logger.LogInformation("Injected appsettings.json file to project");
+logger.LogInformation("Injected secured settings to project");
 Console.WriteLine();
 serviceCollection.AddSingleton<IConfiguration>(configuration);
 logger.LogInformation("Injected configuration");
 Console.WriteLine();
 serviceCollection.AddSingleton<TestClass>();
+var apiKey = configuration.GetSection("DeribitAPI").Get<ServiceClient.ApiKey>();
+if (apiKey != null)
```

Comment:
We should show an error to the user in case _apiKey_ is null and abort, otherwise we will have to check if the _IOptions_ is null.

Any comment on this?

@gsi-emiliogonzalez @gsi-yoancme

Reply:
completely agree